### PR TITLE
Allow overriding of choice quoting [Resolves #81]

### LINKED ISSUE #81: Allow override of choice quoting in Compare
It would be useful to control choice quoting more granularly; for instance, to make a categorical that operates on an array column using infix notation, you must use Compare with choices that look like this:
 
`array['{}'::varchar].format(choice)`

You don't want these to be quoted, and the maybe_quote function will fail this because it is a string.

One option is to give Compare a quote_choices kwarg which overrides the smart quoting.

Another option, that would use the type-aware pattern common within collate, would be to let the caller pass SQLalchemy literals as choices, and to forgo quoting in that case.

### PR DESCRIPTION
**Pull request by thcrock**
I gave the choice quoting override a stab.

### CODE REVIEW

File: collate/collate.py [hunk 1]
```diff
@@ -197,9 +197,9 @@ def get_columns(self, when=None, prefix=None, format_kwargs=None):
             yield ex.literal_column(column).label(to_sql_name(name))
 
 
-def maybequote(elt):
+def maybequote(elt, quote_choices):
```
Comment (mbauman) [inline collate/collate.py]: Maybe call this parameter `always_quote`?  And also have it default to `None`?

File: collate/collate.py [hunk 2]
```diff
@@ -197,9 +197,9 @@ def get_columns(self, when=None, prefix=None, format_kwargs=None):
...
     "Quote for passing to SQL if necessary, based upon the python type"
-    if isinstance(elt, Number):
+    if quote_choices is False or (quote_choices is None and isinstance(elt, Number)):
```
Comment (mbauman) [inline collate/collate.py]: It'd be nice to separate these behaviors a bit more clearly — I'm thinking something like

```
if `always_quote` is None:
    # do the smart-quoting
elif `always_quote`:
    # just put quotes around it
else:
    # return the element
```

File: collate/collate.py (hunk 1 above)
Comment (thcrock) [inline collate/collate.py]: To me it's not clear based on the name that 'always_quote' being equal to False would mean 'never quote'.

Comment (thcrock) [inline collate/collate.py]: I do see that 'choices' doesn't make sense this far down, though. 'quote_override'?

**Comment by mbauman** [comment]
LGTM! Thanks for doing this.

**Pull request merged**